src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -612,6 +770,101 @@ public override void Dispatch(Cci.MetadataVisitor visitor)
         public override Cci.INestedTypeReference AsNestedTypeReference => this;
     }
 
+    /// <summary>
+    /// A type synthesized for each eligible string literal to hold the lazily-initialized string.
+    ///
+    /// https://github.com/dotnet/roslyn/blob/main/docs/features/string-literals-data-section.md
+    /// </summary>
+    internal sealed class DataSectionStringType : DefaultTypeDef, Cci.INestedTypeDefinition
+    {
+        private readonly string _name;
+        private readonly Cci.ITypeReference _systemObject;
+        private readonly Cci.INamedTypeDefinition _containingType;
+        private readonly Cci.IFieldDefinition _field;
+        private readonly ImmutableArray<Cci.IFieldDefinition> _fields;
+        private readonly ImmutableArray<Cci.IMethodDefinition> _methods;
+
+        public DataSectionStringType(
+            ITokenDeferral module,
+            string name,
+            Cci.ITypeReference systemObject,
+            Cci.ITypeReference systemString,
+            Cci.INamespaceTypeDefinition containingType,
+            MappedField dataField,
+            Cci.IMethodDefinition bytesToStringHelper,
+            DiagnosticBag diagnostics)
+        {
+            _name = name;
+            _systemObject = systemObject;
+            _containingType = containingType;
```

Comment:
>_containingType [](http://example.com/codeflow?start=12&length=15)

Would it make sense to use ```PrivateImplementationDetails``` for this field and get rid of the ```_systemObject``` field? Also we could avoid the need to pass ```systemString``` and ```bytesToStringHelper``` explicitly.